rskj-core/src/main/java/co/rsk/peg/federation/FederationSupportImpl.java

```diff
@@ -0,0 +1,713 @@
+     * Logic is as follows:
+     * When no "new" federation is recorded in the blockchain, then return GENESIS
+     * When a "new" federation is present and no "old" federation is present, then return NEW
+     * When both "new" and "old" federations are present, then
+     * 1) If the "new" federation is at least bridgeConstants::getFederationActivationAge() blocks old,
+     * return the NEW
+     * 2) Otherwise, return OLD
+     *
+     * @return a reference to where the currently active federation is stored.
+     */
+    private StorageFederationReference getActiveFederationReference() {
+        Federation newFederation = provider.getNewFederation(constants, activations);
+
+        // No new federation in place, then the active federation
+        // is the genesis federation
+        if (newFederation == null) {
+            return StorageFederationReference.GENESIS;
+        }
+
+        Federation oldFederation = provider.getOldFederation(constants, activations);
+
+        // No old federation in place, then the active federation
+        // is the new federation
+        if (oldFederation == null) {
+            return StorageFederationReference.NEW;
+        }
+
+        // Both new and old federations in place
+        // If the minimum age has gone by for the new federation's
+        // activation, then that federation is the currently active.
+        // Otherwise, the old federation is still the currently active.
+        if (shouldFederationBeActive(newFederation)) {
+            return StorageFederationReference.NEW;
+        }
+
+        return StorageFederationReference.OLD;
+    }
+    private boolean shouldFederationBeActive(Federation federation) {
+        long federationAge = rskExecutionBlock.getNumber() - federation.getCreationBlockNumber();
+        return federationAge >= constants.getFederationActivationAge(activations);
+    }
+    private Federation getGenesisFederation() {
+        long GENESIS_FEDERATION_CREATION_BLOCK_NUMBER = 1L;
+        List<BtcECKey> genesisFederationPublicKeys = constants.getGenesisFederationPublicKeys();
+        List<FederationMember> federationMembers = FederationMember.getFederationMembersFromKeys(genesisFederationPublicKeys);
+        Instant genesisFederationCreationTime = constants.getGenesisFederationCreationTime();
+        FederationArgs federationArgs = new FederationArgs(federationMembers, genesisFederationCreationTime, GENESIS_FEDERATION_CREATION_BLOCK_NUMBER, constants.getBtcParams());
+        return FederationFactory.buildStandardMultiSigFederation(federationArgs);
+    }
+
+    @Override
+    public Optional<Script> getActiveFederationRedeemScript() {
+        return activations.isActive(RSKIP293) ?
+            Optional.of(getActiveFederation().getRedeemScript()) :
+            Optional.empty();
+    }
+
+    @Override
+    public Address getActiveFederationAddress() {
+        return getActiveFederation().getAddress();
```

Comment:
Looks like getActiveFederation is called many times. Should we have a class instance of the activeFederation?

```
Federation getActiveFederation() {
   if (activeFederation == null) {
     ....
   }
  return activeFederation;
```

Guess there's no risk that the active federation changes during the use of this class, is it?